src/main/java/cashleh/parser/StringTokenizer.java

```diff
@@ -30,6 +30,7 @@ private static void checkAllPrefixesPresent(String input, String[] prefixes) thr
         for (int i = 0; i < prefixes.length; i++) {
             if (prefixes[i].contains(":optional")) {
                 prefixes[i] = prefixes[i].replace(":optional", "");
+                assert(!prefixes[i].contains(":optional"));
```

Comment:
When will this assertion be false ah? When somehow accidentally the string got more than one ":optional"?

Reply:
i dont think assertions are supposed to be false